plugins/techdocs/src/home/components/actions.tsx

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import ShareIcon from '@material-ui/icons/Share';
+import { DocsTableRow } from './types';
+
+export function createCopyDocsUrlAction(copyToClipboard: Function) {
+  return (row: DocsTableRow) => {
+    return {
+      icon: () => <ShareIcon fontSize="small" />,
+      tooltip: 'Click to copy documentation link to clipboard',
+      onClick: () =>
+        copyToClipboard(
+          `${window.location.href.split('/?')[0]}/${row.resolved.docsUrl}`,
```

Comment:
```suggestion
          `${window.location.href.split('\?')[0]}/${row.resolved.docsUrl}`,
```
Hmm, does this look more correct? I'm assuming the intent is to remove the query params here.

Reply:
hm did you mean to use a regex ? `.split(/\?\)`?
I think we want to remove the `/` too if there are query params so we can always appropriately append a `/` before `docsUrl` and handles both scenarios:
- `'backstage.com/docs/?foo=bar'` => `'backstage.com/docs'` + `'/'` + `docsUrl`
  - if we only remove everything after `'?'` then we end up with an extra `'/'` if we append it
- `'backstage.com/docs` => `'backstage.com/docs'` + `'/'` + `docsUrl`
  - if we split by `'?'` only and don't append a `'/'` (to support the first scenario) then the url is malformed

Let me know if I misunderstood something

Reply:
Ah, got it -- I was testing with `backstage.com/docs?foo=bar` and `/?` didn't strip anything in this case. If a `/` will always be present on the URL here then nothing to worry about. Otherwise it could be `.split(/\/?\?/)` to make the slash optional 🤔

Reply:
good point! I didn't consider the scenario where a trailing / is not present, it looks like it currently always adds it for query params but I updated to use the optional matching regex just in case, thanks!

Reply:
hmm string operations on URLs... May I suggest to do `new URL(docsUrl)` and concatenate the fields of it that you need (just skip the query part). This could all also go into a nicely named separate helper function that clarifies what's being done

Reply:
There's also the two-argument form of the URL constructor that does relative resolution, but it also has quirks (the base argument should have a trailing slash), plus not sure it applies here

Reply:
Thanks @freben for the suggestion! I ultimately settled on the following implementation: 
```ts
`${window.location.origin}${window.location.pathname.replace(/\/?$/, '/')}${row.resolved.docsUrl}`
```

This should more pragmatically remove the query params and append the `docsUrl` path while robustly handling inserting the `/` separator when necessary. Let me know if there anything else I missed!